### PR TITLE
Detect if song is instrumental

### PR DESCRIPTION
**Pull request by JamesVStone**
Uses spotify `audio-features` API to determine if a song actually has lyrics. This should stop unnecessary unsupported issues being added to GH.

### CODE REVIEW

**Review by aadibajpai — CHANGES_REQUESTED** [review]
there appear to be merge conflicts

**Comment by aadibajpai** [comment]
I think it happens because your branch is 12 commits behind master

**Comment by JamesVStone** [comment]
It doesn't let me click the resolve conflict button. Will I have to do this manually?

**Comment by aadibajpai** [comment]
> It doesn't let me click the resolve conflict button. Will I have to do this manually?

I think so, maybe google something 🤔

**Comment by JamesVStone** [comment]
Considering the report i'll add some tests to cover the logic which I have written. All conflicts are resolved though :smile:

**Comment by JamesVStone** [comment]
should I be using my own dev branch then squashing to my master, then to pull request?

File: swaglyrics_backend/issue_maker.py [hunk 1]
```diff
@@ -234,12 +239,18 @@ def check_song(song, artist):
...
         return False
     if data:
-        print(data[0]['artists'][0]['name'], data[0]['name'])
-        if data[0]['name'] == song and data[0]['artists'][0]['name'] == artist:
+        track = data[0]
+        print(track['artists'][0]['name'], track['name'])
+        if track['name'] == song and track['artists'][0]['name'] == artist:
             print(f'{song} and {artist} legit on Spotify')
+            metadata = requests.get(f'https://api.spotify.com/v1/audio-features/{track["id"]}', headers=headers)
+            if metadata["instrumentalness"] > 0.45 and metadata["speechiness"] < 0.04:
```
Comment (aadibajpai) [inline swaglyrics_backend/issue_maker.py]: I thought the bar for instrumentalness was 0.5 according to the API docs?

Comment (JamesVStone) [reply swaglyrics_backend/issue_maker.py]: I found that there was some overlap. In my experiment this was the optimum value. Each of the attributes has a different scale which is why the threshold for speechiness is so low.

**Comment by aadibajpai** [comment]
> Considering the report i'll add some tests to cover the logic which I have written. All conflicts are resolved though 😄

Yep that would be cool

**Comment by aadibajpai** [comment]
> should I be using my own dev branch then squashing to my master, then to pull request?

Just push to your branch

File: swaglyrics_backend/issue_maker.py (hunk 1 above)
Comment (aadibajpai) [reply swaglyrics_backend/issue_maker.py]: Okay, no problem then

File: swaglyrics_backend/issue_maker.py [hunk 2]
```diff
@@ -234,12 +239,18 @@ def check_song(song, artist):
     except KeyError:
         return False
     if data:
-        print(data[0]['artists'][0]['name'], data[0]['name'])
-        if data[0]['name'] == song and data[0]['artists'][0]['name'] == artist:
+        track = data[0]
+        print(track['artists'][0]['name'], track['name'])
+        if track['name'] == song and track['artists'][0]['name'] == artist:
             print(f'{song} and {artist} legit on Spotify')
+            metadata = requests.get(f'https://api.spotify.com/v1/audio-features/{track["id"]}', headers=headers).json()
```
Comment (JamesVStone) [inline swaglyrics_backend/issue_maker.py]: @aadibajpai @flabbet  how can I mock the two get requests separately? I need to return different responses and it is only patched once to return `Response()`.

Comment (flabbet) [reply swaglyrics_backend/issue_maker.py]: You can separate tests, you don't have to write two cases in one test

Comment (aadibajpai) [reply swaglyrics_backend/issue_maker.py]: Also for later reference, use side_effect, there are some examples in the main SwagLyrics tests

**Comment by aadibajpai** [comment]
So it looks good but the only thing I feel we wanna do is if the song is instrumental then append a message to the user. I'm trying to figure out how to incorporate that into the flow.

**Comment by JamesVStone** [comment]
> So it looks good but the only thing I feel we wanna do is if the song is instrumental then append a message to the user. I'm trying to figure out how to incorporate that into the flow.

Would that require modifications to the client.

**Comment by aadibajpai** [comment]
> > So it looks good but the only thing I feel we wanna do is if the song is instrumental then append a message to the user. I'm trying to figure out how to incorporate that into the flow.
> 
> Would that require modifications to the client.

Nope there's a parameter passed to the client that's appended when lyrics aren't fetched

**Review by aadibajpai — APPROVED** [review]

**Pull request merged**